src/app/_components/Header/Header.js

```diff
@@ -1,9 +1,29 @@
-import React from 'react';
+'use client';
+
+import React, { useState } from 'react';
+import Modal from '../Modal';
 
 export default function Header() {
```

Comment:
NIT(Just advice): default exports are bad practice in general, cuz can let you `import Foo from 'bar'` and vice-versa. Meanwhile named export will prevent such cases (`import {Foo} from 'foo'`). Try to use default exports only if that's really necessary.